sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -215,22 +233,35 @@ public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse
      * It will call cancelOperation if status is {@link OperationStatus#IN_PROGRESS} otherwise it does nothing.
      *
      * @throws UnsupportedOperationException when the cancel operation is not supported by the Azure service.
+     * @return A {@link Mono} containing the poller response.
      */
-    public void cancelOperation() throws UnsupportedOperationException {
+    public Mono<T> cancelOperation() throws UnsupportedOperationException {
         if (this.cancelOperation == null) {
-            throw logger.logExceptionAsError(new UnsupportedOperationException(
-                "Cancel operation is not supported on this service/resource."));
+            return Mono.error(logger.logExceptionAsError(new UnsupportedOperationException(
+                "Cancel operation is not supported on this service/resource.")));
         }
 
         // We can not cancel an operation if it was never started
         // It only make sense to call cancel operation if current status IN_PROGRESS.
+
         final PollResponse<T> response = this.pollResponse;
         if (response != null && response.getStatus() != OperationStatus.IN_PROGRESS) {
-            return;
+            return Mono.empty();
         }
 
-        // Time to call cancel
-        this.cancelOperation.accept(this);
+        //Time to call cancel
+        return this.cancelOperation.apply(this);
+    }
+
+    /**
+     * Returns the final result if the polling operation has been completed.
+     * @return A {@link Mono} containing the final output.
+     */
+    public Mono<R> result() {
```

Comment:
Why isn't this getResult()?

Reply:
valid point, 
added get prefix.

Reply:
the get prefix will take some features to become a habit.